zerver/tests/test_bots.py

```diff
@@ -146,7 +146,7 @@ def test_json_users_with_bots(self) -> None:
 
         self.assert_json_success(users_result)
 
-        self.assert_length(queries, 3)
+        self.assert_length(queries, 9)
```

Comment:
This change is wrong; the fact that this number is going from 3 to 9 means you have database queries in a loop.

A quick print-debug suggests it's UserProfile fetches.

Reply:
4 of them if I remember. I was concerned about this too.